src/adapter/ads/nativeads.js

```diff
@@ -27,25 +27,27 @@ let NativeAdsAdapter = youbora.Adapter.extend({
 
   /**  @returns {String} - current ad position (only ads) */
   getPosition: function () {
-    let returnValue = youbora.Adapter.AdPosition.MIDROLL;
-    switch (this.adPosition) {
-      case 'preroll':
-        returnValue = youbora.Adapter.AdPosition.PREROLL;
-        break;
-      case 'postroll':
-        returnValue = youbora.Adapter.AdPosition.POSTROLL;
-        break;
-      case 'midroll':
-        break;
-      case 'overlay':
-        returnValue = 'overlay';
-        break;
-      default:
-        if (!this.plugin.getAdapter().flags.isJoined) {
-          returnValue = youbora.Adapter.AdPosition.PREROLL;
-        } else if (!this.plugin.getAdapter().getIsLive() && this.plugin.getAdapter().getPlayhead() > this.plugin.getAdapter().getDuration() - 1) {
-          returnValue = youbora.Adapter.AdPosition.POSTROLL;
-        }
+    let returnValue = youbora.Constants.AdPosition.Midroll;
+    if (this.adPosition) {
+      switch (this.adPosition) {
+        case 'preroll':
+          returnValue = youbora.Constants.AdPosition.Preroll;
+          break;
+        case 'postroll':
+          returnValue = youbora.Constants.AdPosition.Postroll;
+          break;
+        case 'midroll':
+          break;
+        case 'overlay':
+          returnValue = 'overlay';
+          break;
+      }
+    } else {
+      if (!this.plugin.getAdapter().flags.isJoined) {
+        returnValue = youbora.Constants.AdPosition.Preroll;
+      } else if (!this.plugin.getIsLive() && this.plugin.getPlayhead() > this.plugin.getDuration() - 1) {
```

Comment:
Why did you remove `getAdapter()` here? is it related to this issue?

Reply:
No is not, now the object plugin is accessing to the adapter methods instead (safely, checking that both the plugin and the method exist) so we can save some code while adding an extra check